test/functional/p2p_v2_misbehaving.py

```diff
@@ -166,7 +167,7 @@ def test_v2disconnection(self):
                 node0.add_p2p_connection(MisbehavingV2Peer(test_type), wait_for_verack=True, send_version=True, supports_v2_p2p=True)
                 self.log.info(f"No disconnection for {test_type.name}")
             else:
-                with node0.assert_debug_log(expected_debug_message[test_type.value], timeout=5):
+                with node0.assert_debug_log(expected_debug_message[test_type.value], timeout=10):
```

Comment:
> Ok, I see, the peer.wait_for_disconnect() immediately returns, as there is no connection yet, and doesn't do anything.


I meant it as a check so that we know if `self._transport` got reset to `None` at some point of time. (`self._transport` is set in `connection_made()` when the connection is opened irrespective of whether p2p connection remains connected or disconnected)

> Why is it required to set a timeout here at all? The log message should happen before the disconnect, which is waited on, no?

Hmm yes, makes sense that we don't need a timeout.